images/virtualization-artifact/pkg/controller/vmip/internal/iplease_handler.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8snet "k8s.io/utils/net"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/deckhouse/virtualization-controller/pkg/controller/common"
+	"github.com/deckhouse/virtualization-controller/pkg/controller/vmip/internal/state"
+	"github.com/deckhouse/virtualization-controller/pkg/controller/vmip/internal/util"
+	virtv2 "github.com/deckhouse/virtualization/api/core/v1alpha2"
+)
+
+const IpLeaseHandlerName = "IPLeaseHandler"
+
+type IPLeaseHandler struct {
+	client      client.Client
+	logger      logr.Logger
+	ParsedCIDRs []*net.IPNet
```

Comment:
I think we should use []net.Ipnet without pointers. In the constructor we can dereference

Reply:
In func 'allocate New IP' used 'k8snet.Range Size', which apply 'subnet *net.IP Net'.